src/hooks/useBridgedBlocks.ts

```diff
@@ -36,32 +37,53 @@ interface Props {
 }
 
 const useBridgedBlocks = ({ isApiReady, api, chain }: Props) => {
-  const [bestBridgedFinalizedBlock, setBestBridgedFinalizedBlock] = useState('');
-
+  const [bestBridgedFinalizedBlock, setBestBridgedFinalizedBlock] = useMountedState('');
+  const [bestFinalizedBlock, setBestFinalizedBlock] = useMountedState('');
   const { bridgedGrandpaChain } = getSubstrateDynamicNames(chain);
+
+  useEffect((): (() => void) | undefined => {
+    let unsubBestFinalized: Promise<VoidFn>;
+
+    const shouldProceed: boolean = !!(isApiReady && chain);
+    if (shouldProceed) {
+      unsubBestFinalized = api.query[bridgedGrandpaChain].bestFinalized((res: CodecHeaderId) => {
+        const bestFinalized = res.toString();
+        setBestFinalizedBlock(bestFinalized);
+      });
+    }
+
+    return () => {
+      unsubBestFinalized
+        .then((u) => {
+          u();
+        })
+        .catch((e) => logger.error('error unsubscribing unsubBestFinalized', e));
+    };
+  }, [isApiReady, chain, api, bridgedGrandpaChain, setBestFinalizedBlock]);
+
   useEffect((): (() => void) | undefined => {
-    const shouldProceed: boolean = !!(api && isApiReady && chain);
-    if (!shouldProceed) return;
+    let unsubImportedHeaders: Promise<VoidFn>;
+    const shouldProceed: boolean = !!(api && isApiReady && chain && bestFinalizedBlock);
```

Comment:
Repeating the condition sucks, IMHO it was much cleaner previously in a single `useEffect.

Reply:
> What about if i extract the shouldProceed outside useEffects and i make the useEffect to depend on it? should that be a solution?

Yes, that's a much better solution:
1. No code duplication
2. Less dependencies to `useEffect`.

I still think that one `useEffect` would be easier to understand here. These two effecst are not independent - the second one depends on the first one and these kind of dependencides are hard to understand if `useEffects` are separate - you basically need to go through the entire flow of "if new block is updated, then this other things gets fired, which updates the `importedHeaders`".